src/components/map/Map.js

```diff
@@ -39,9 +39,9 @@ const Map = ({
         lng={cluster.lng}
       />
     ))}
-    {locations.map((location) => (
+    {locations.map((location, index) => (
       <Location
-        key={location.id}
+        key={index}
```

Comment:
I'm getting some really weird issues with the rendering on Safari. This is good to look into at the beginning of today's dev night (on Safari, just zoom in and out of somewhere and things will go out of wack.)

I'm not sure if it's related to this or not, but could you try using the id from the location as a key instead? In general it's not good practice to use an index as the key. https://reactjs.org/docs/lists-and-keys.html

It's best to add a key to Cluster as well. For that, try writing a helper function to "hash" the longitude and latitude.